lang/java/protobuf/src/main/java/org/apache/avro/protobuf/ProtobufData.java

```diff
@@ -191,26 +192,27 @@ public Schema getSchema(Class c) {
     return schema;
   }
 
-  private static final ThreadLocal<Map<Descriptor, Schema>> SEEN = ThreadLocal.withInitial(IdentityHashMap::new);
+  private static final ThreadLocal<Map<DescriptorProto, Schema>> SEEN = ThreadLocal.withInitial(IdentityHashMap::new);
 
   public Schema getSchema(Descriptor descriptor) {
-    Map<Descriptor, Schema> seen = SEEN.get();
-    if (seen.containsKey(descriptor)) // stop recursion
-      return seen.get(descriptor);
+    Map<DescriptorProto, Schema> seen = SEEN.get();
+    DescriptorProto dp = descriptor.toProto();
+    if (seen.containsKey(dp)) // stop recursion
+      return seen.get(dp);
```

Comment:
Would it be better to do a `.get()` + check for `null` instead of two lookups (`containsKey()` + `get()`) ?

Reply:
Makes sense, I didn't want to change too much of the current implementation though.